modules/openapi-generator/src/main/resources/JavaSpring/paramDoc.mustache

```diff
@@ -0,0 +1 @@
+{{#oas3}}@Parameter(name = "{{{baseName}}}", description = "{{{description}}}"{{#required}}, required = true{{/required}}, schema = @Schema({{> allowableValues }}{{^isContainer}}{{#defaultValue}}, defaultValue = "{{{.}}}"{{/defaultValue}}{{/isContainer}})){{/oas3}}{{^oas3}}@ApiParam(value = "{{{description}}}"{{#required}}, required = true{{/required}}{{#allowableValues}}, {{> allowableValues }}{{/allowableValues}}{{^isContainer}}{{#defaultValue}}, defaultValue = "{{{.}}}"{{/defaultValue}}{{/isContainer}}){{/oas3}}
```

Comment:
If there's no allowable values defined or defaults, does `@Schema()` work as an annotation?

Also, I this removes the `defaultValue` applied to the `@Parameter` annotation (one level up) - does that still work?

Reply:
> If there's no allowable values defined or defaults, does @Schema() work as an annotation?

An empty `@Schema()` is the default anyway, Not pretty, but ok.

> Also, I this removes the defaultValue applied to the @Parameter annotation (one level up) - does that still work?

`@Parameter`  hast no defaultValue. IIRC. This one of the bugs introduced. `defaultValue` and `allowableValues`  moved to `@Schema`.

But I identified another problem. Maybee somone more experienced in mustache than I can help. When `allowableValues` is empty, but `defaultValue` is set. `schema = @Schema(, defaultValue = "11.2"))` is generated, which will obviously not compile. How to deal with the comma when there is nota single value of Schema which is always set?

Reply:
Move the `defaultValue` first. If it's set, good, if not, then make it `defaultValue = ""`, which is the default there. After that, if the `allowableValue` is set, it can safely have the comma before it.

Reply:
I actually used `description=""` because defaultValue is not used at all times. It's usage is constrained by the `isContainer`  flag.